### PR TITLE
[JENKINS-38302] Blue Ocean compatiblity

### PR DESCRIPTION
**Pull request by amuniz**
This is adding BO compatibility and also adding some of the features already offered in GH Branch source (specific views for branches and PRs and some customized texts).

@kzantow @i386 can you test this PR and provide feedback before merging?

### CODE REVIEW

**Comment by i386** [comment]
So cool! I'd be happy to give this a test on Monday :)

**Comment by i386** [comment]
What would you say the large difference is between this and Github branch source after this change?

**Comment by amuniz** [comment]
@i386 

> What would you say the large difference is between this and Github branch source after this change?

[JENKINS-36283](https://issues.jenkins-ci.org/browse/JENKINS-36283), but that's not affecting BO I think.

**Comment by amuniz** [comment]
@scherler @kzantow I've found a glitch: if you click on a PR row, a blank page is shown, do you think is something to fix at this side or at BO side?

**Comment by kzantow** [comment]
@amuniz it's a bug in BO: https://issues.jenkins-ci.org/browse/JENKINS-38300

File: src/main/resources/com/cloudbees/jenkins/plugins/bitbucket/AbstractBranchJobFilter/config.jelly [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<!-- not visible in UI -->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form" />
```
Comment (kzantow) [inline src/main/resources/com/cloudbees/jenkins/plugins/bitbucket/AbstractBranchJobFilter/config.jelly]: Is this file needed? I'm assuming so...

**Comment by kzantow** [comment]
:bee: AFAICT, this is awesome! :)

**Comment by amuniz** [comment]
Merged into master for other reasons, but some manual tests are still welcome @i386 @scherler @kzantow 

Not releasing yet.

**Comment by i386** [comment]
@amuniz will give this a test in a little while. Mostly want to ensure that it is comparable to Github in functionality before speeding on it in the aggregator plugin for Blue Ocean 🌊
